pkg/apis/hive/v1/clusterdeployment_types.go

```diff
@@ -345,6 +350,11 @@ type ControlPlaneConfigSpec struct {
 	// ServingCertificates specifies serving certificates for the control plane
 	// +optional
 	ServingCertificates ControlPlaneServingCertificateSpec `json:"servingCertificates,omitempty"`
+
+	// APIURLOverride is the URL that Hive will use to communicate with the API server
+	// of the remote cluster.
```

Comment:
Suggest something like: APIURLOverride is the optional URL override that Hive will transition to communicate with the API server of the remote cluster, once we are able to determine it is active.